### ISSUE TITLE
Shorten Classifications menu for easier access

### ISSUE THREAD

**Issue by lacan** [qupath/qupath#1644]
The most common things to do in the classifications list in my experience is to
1. Create or delete classes
2. Show or hide classes

One option for Show/Hide would be to have the right click menu be context sensitive:
- If one or multiple classifications are selected and visible, the context menu becomes "Hide selected classes" (Instead of show/Hide > Hide classes in viewer)
- If one or multiple classifications are selected and hidden, the context menu becomes "Show selected classes" (Instead of show/Hide > Show classes in viewer)

One option for Add/remove would be that the right click menu:
- Includes an "Add new classification" at the root of the menu
- If one or multiple classifications are selected, change the menu to include "Remove selected classifications" (Instead of Add/remove > Remove class, which is a slight misnomer as it's possible to remove multiple classes if they are selected)

While I know that there is the <kbd>Spacebar</kbd> shortcut for the Hide/Show classes that is way more used, I thought I'd suggest these changes, especially for adding classifications.

Best

Oli

**Comment by petebankhead** [comment]
> One option for Show/Hide would be to have the right click menu be context sensitive:
> 
> * If one or multiple classifications are selected and visible, the context menu becomes "Hide selected classes" (Instead of show/Hide > Hide classes in viewer)
> * If one or multiple classifications are selected and hidden, the context menu becomes "Show selected classes" (Instead of show/Hide > Show classes in viewer)

What about if multiple classifications are selected, some shown and some hidden?

I'm apprehensive about trying to make the code too clever, and confusing people more by the menu changing when they click it.

If the core issue is that the common things are too hard to access, would simply moving `Show/hide` higher up the menu be a solution?

My hesitation with that is that currently the top part of the menu (above the divider) is all concerned with adding/removing classifications. So `Show/Hidez would have to go to the *very* top, to avoid interrupting this logical grouping... and, as you say, spacebar does that job, so `Add/Remove` might be needed more often.

**Comment by lacan** [comment]
> I'm apprehensive about trying to make the code too clever, and confusing people more by the menu changing when they click it.

Yeah that makes sense! Maybe context sensitive isn't the way to go. Thank you for considering it though! :)

> If the core issue is that the common things are too hard to access, would simply moving Show/hide higher up the menu be a solution?

This could definitely work as well. I would be super happy to have a one menu access for "Add/Remove", in the same vibe as "Estimate stain vectors" was given its own high-leven location instead of a submenu.

**Comment by petebankhead** [comment]
This has now been addressed through
* https://github.com/qupath/qupath/pull/1752
* https://github.com/qupath/qupath/pull/1756
